native/dlartg.go

```diff
@@ -17,7 +17,8 @@ import "math"
 func (impl Implementation) Dlartg(f, g float64) (cs, sn, r float64) {
 	safmin := dlamchS
 	eps := dlamchE
-	safmn2 := math.Pow(float64(dlamchB), math.Log(safmin/eps))
+	exp := int(math.Log(safmin/eps) / math.Log(float64(dlamchB)) / 2)
+	safmn2 := math.Pow(float64(dlamchB), float64(exp))
```

Comment:
With the simplified form it's clearer what the formula does, it's not so magical:
```
Log(safmin/eps) / Log(dlamchB) / 2 = Log_{dlamchB}(safmin/eps) / 2
                                   = Log_{dlamchB}(Sqrt(safmin/eps))
```
Using that expression as an exponent for `dlamchB`:
```
dlamchB^{Log_{dlamchB}(Sqrt(safmin/eps))} = Sqrt(safmin/eps))
```
So basically
```
safmn2 := math.Sqrt(safmin/eps))
```
I write "basically" because above I omitted the `INT` function used in Fortran. I don't know whether including it is important or not. Also note that `math.Floor` is not the same as `float64(int())` for negative numbers, which is the case here.
I cannot explain completely why they compute `safmn2` as they do. I think that the square is there to make sure that the numbers don't get too small when `f1*f1` or `g1*g1` are computed. I don't know why divide by eps, and also why go through the exponentiation of `dlamchB`. The only reason to do that seems to be the possibility to use the `INT` function whose objective is to make the exponentiation in the form `2^{negative integer}`. Again, I don't know why.